scripts/apps/search/services/SearchService.js

```diff
@@ -592,6 +604,19 @@ export function SearchService($location, gettext, config, session) {
     };
 
     /**
+     * @ngdoc method
+     * @name search#doesSearchAgainstRepo
+     * @public
+     * @return {bool}
+     * @description Checks if the given search object will do the search agains the given repo
+     * @param {Object} search search criteria
+     * @param {string} repo name of the repo: ingest, archive, published, archived
```

Comment:
Here, I used to capitalise `String`, but probably yours is more correct since it reflects the output of `typeof` in JS. I guess it's fine to keep it like this.